### ISSUE TITLE
Streaming Recognition with Longer Videos(5+ mins)

### ISSUE THREAD

**Issue by ibrahimdevs** [felixjunghans/google_speech#22]
Hi @felixjunghans and @sveinbjornt , firstly thanks for great plugin. We saved lots for time thanks to this plugin.

We are using this plugin with Streaming Recognition. As I know we can use it for up to 3 min. videos and 5 min videos with beta version. I have to support 10 min videos, is there any solution or workaround with Streaming Recognition?

I saw "Add infinity stream support" your TODO list, is it the only answer that I'm looking for?

Thanks,

**Comment by ibrahimdevs** [comment]
@felixjunghans you closed this issue but is there any way to support longer streaming recognition? As far as I know there is no update for that issue.

**Comment by felixjunghans** [comment]
Sorry closed by mistake.

**Comment by alaincruz06** [comment]
Hello @felixjunghans & @ibrahimdevs I saw this article yesterday: [How to get rid of Longtime elapsed without audio and Resource exhausted error in Google Speech to Text for m3u8 video stream](https://madhura-gore.medium.com/how-to-get-rid-of-longtime-elapsed-without-audio-and-resource-exhausted-error-in-google-speech-to-a471c47fdec) with Node.js and I think I came up with a solution, at least it's working for me:
Note: I'll post the translations I did from JS to dart:

To set the max. amount of streaming time allowed
```
//From Node 
let streamingLimit = 210000 //3.5 minutes
//To Dart
const int kstreamingLimit = 210000;
```

Restarting stream every 3.5 mins for infinite streaming
```
//From Node 
timeout = setTimeout(restartStream, streamingLimit);
//To Dart
 Future.delayed(const Duration(seconds: kstreamingLimit), restartStream);
```

I used the previous function at the end of  
```
responseStream.listen((data) {
//logic to listen to data and react accordingly 
....


 Future.delayed(const Duration(seconds: kstreamingLimit), restartStream);
}
```

and then
```
//From Node 
function restartStream() {
    if (recognizeStream) {
        recognizeStream.removeListener('data', speechCallback);
        recognizeStream.destroy();
        recognizeStream = null;
    }
}
//To Dart  
void restartStream() {
    if (responseStream != null) {
      responseStream!.drain();
      responseStream = null;
    }
  }
```

and it's working for me so far...hope it helps.

**Comment by felixjunghans** [comment]
Hi @alaincruz06 thanks for your idea. I had something similar in mind but do you have any experience what happens if the stream has to be restarted in the middle of a sentence or even a word? Does Google still recognize it then?

**Comment by alaincruz06** [comment]
I guess it's all about what you do in your "listen" method. Either you restarted or not the service (the stream) it keeps listening until you stop recording.
In that article the author also uses a workaround to send 0's to the stream and in case the stream is silent, so we don't get "Long time elapsed without audio" Exception
I should clarify I'm talking what I just believe, I'm not sure about the official Google logic.
There's a similar workaround in [speech_to_text](https://pub.dev/packages/speech_to_text) package that I used to restart the service and it worked without overwritting my text.

**Comment by ibrahimdevs** [comment]
@felixjunghans any update about this feature? I'm excited to see some new version for this package, but none of them is about endless streaming :)

**Comment by felixjunghans** [comment]
@ibrahimdevs 
@alaincruz06 


I have just released version 5.0.0 with endless streaming support. Please test if this solves your problem.
